src/js/d3Utils.js

```diff
@@ -23,6 +23,42 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         return d3.selectAll(elem.toArray());
     };
 
+    /* Given the object "d", returns value of:
+     * - "id" if it exists on the top-level object,
+     * - "data.id" if it exists on a "data" object
+     * - "undefined" if neither of these exist
+     */
+    floe.d3.idExtractor = function (d) {
+        try {
+            var id = d.id !== undefined ? d.id : d.data.id;
+            return id !== undefined ? id : undefined;
+        } catch (e) {
```

Comment:
It was an awkward construction, but actually necessary to avoid a possible error when trying to access d.data.id (d.data might not exist). I restructured into a series of ternary operators so try/catch isn't necessary.